src/views/CenterPaneView.tsx

```diff
@@ -1,39 +1,46 @@
 import { FileEntry } from '@tauri-apps/api/fs';
 import { useEffect, useState } from 'react';
 
+import { FileEntryInfo } from '../features/openedFiles/openedFilesSlice';
 import { readFile } from '../filesystem';
+import { useAppSelector } from '../redux/hooks';
 import { TipTapEditor } from '../TipTapEditor/TipTapEditor';
-import { CenterPaneViewProps } from '../types/CenterPaneViewProps';
+import { EditorAPI } from '../types/EditorAPI';
 
-function isTipTap(file?: FileEntry) {
-  return file?.path.endsWith('.tiptap');
+export interface CenterPaneViewProps {
+  editorRef: React.MutableRefObject<EditorAPI | undefined>;
 }
 
-export function CenterPaneView({ file, ...props }: CenterPaneViewProps) {
-  const [loading, setLoading] = useState(true);
+export function CenterPaneView(props: CenterPaneViewProps) {
+  const selectedFile = useAppSelector((state) => state.openedFiles.selectedFile);
+
+  if (!selectedFile) return <span>Please select a file in the project tree.</span>;
+  if (selectedFile.type === 'TipTap') return <EditorView selectedFile={selectedFile} {...props} />;
+  if (selectedFile.type === 'PDF') return <PDFView file={selectedFile.entry} />;
+  return <span>Unknown file: {selectedFile.entry.path}</span>;
+}
+
+function EditorView({ selectedFile, ...props }: { selectedFile: FileEntryInfo } & CenterPaneViewProps) {
+  const [loading, setLoading] = useState(false);
   const [content, setContent] = useState<string | null>(null);
 
   useEffect(() => {
-    if (file && isTipTap(file)) {
-      setLoading(true);
-      (async () => {
-        const content = await readFile(file);
-        const textContent = new TextDecoder('utf-8').decode(content);
-        setContent(textContent);
-        setLoading(false);
-      })();
-    }
-  }, [file]);
-
-  if (loading)
-    return (
-      <div>
-        <strong>Loading...</strong>
-      </div>
-    );
+    setLoading(true);
+    (async () => {
+      // Note: Somehow I need need to await a bit for the TipTap to work!!
+      await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
This is something we should probably investigate further. Maybe you could create an issue about it.

Reply:
You are right. I will create an issue for this.